### DISCUSSION TITLE
Is there any downside or conflict arising from enabling newer C++ features in external scanners?

### DISCUSSION THREAD

**Discussion by ahelwer** [tree-sitter/tree-sitter#1245] [Q&A]
Running into this issue now that I'm trying to build my grammar on non-windows platforms using github actions. C++11 features like auto or initializer lists aren't enabled by default on mac, which I guess uses clang. I don't know whether this is due to something set within the grammar's build process or just a clang default. Is it okay to add the line `"cflags_cc": ["-std=c++11"]` to my binding.gyp file? I ask because the file also includes `"cflags_c": ["-std=c99"]`. I guess this would preclude anyone with a decade-out-of-date compiler from using my grammar?

Edit: actually I just realized that the error comes not with the binding.gyp build from `npm install`, but from the parser compilation when running `tree-sitter test`. I guess the C++ version is locked to something earlier than C++11 in the tree-sitter-cli binary?

Edit2: I opened an issue https://github.com/tree-sitter/tree-sitter/issues/1246

**Comment by sogaiu** [comment]
Presumably this was closed because C is the only language supported for implementing external scanners now.
